### PR TITLE
Allow case lists to handle actions

### PR DESCRIPTION
**Pull request by wpride**
cross-request: https://github.com/dimagi/formplayer/pull/282

fixes: http://manage.dimagi.com/default.asp?246634

cc @czue

### CODE REVIEW

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js [hunk 1]
```diff
@@ -208,6 +208,12 @@ FormplayerFrontend.module("Menus.Views", function (Views, FormplayerFrontend, Ba
 
         templateHelpers: function () {
             var appId = Util.currentUrlToObject().appId;
+            var tempData = this.options.model.get('data');
+            for (var i =0; i < tempData.length; i++) {
+                if(this.options.styles[i] === "graph") {
+                    tempData[i] = tempData[i][1];
+                }
+            }
```
Comment (benrudolph) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js]: this is pretty confusing, where is tempData used and why do we need to do this for just the graph style?

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js [hunk 2]
```diff
@@ -208,6 +208,12 @@ FormplayerFrontend.module("Menus.Views", function (Views, FormplayerFrontend, Ba
 
         templateHelpers: function () {
             var appId = Util.currentUrlToObject().appId;
+            var tempData = this.options.model.get('data');
+            for (var i =0; i < tempData.length; i++) {
+                if(this.options.styles[i] === "graph") {
```
Comment (benrudolph) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js]: `if (`

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js [hunk 3]
```diff
@@ -208,6 +208,12 @@ FormplayerFrontend.module("Menus.Views", function (Views, FormplayerFrontend, Ba
 
         templateHelpers: function () {
             var appId = Util.currentUrlToObject().appId;
+            var tempData = this.options.model.get('data');
+            for (var i =0; i < tempData.length; i++) {
```
Comment (benrudolph) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js]: `i = 0`

File: corehq/apps/cloudcare/templates/formplayer/case_list.html [hunk 4]
```diff
@@ -46,10 +46,12 @@
...
       <% } %>
 
-      <% if (action) { %>
-        <div class="caselist-action-button">
-          <button type="button" class="btn btn-success btn-lg" id="double-management"><%= action.text %></button>
-        </div>
+      <% if (actions) { %>
+        <% _.each(actions, function(action, index) { %>
+            <div class="caselist-action-button">
+              <button type="button" class="btn btn-success btn-lg" id="double-management" index="<%= index %>"><%= action.text %></button>
```
Comment (benrudolph) [inline corehq/apps/cloudcare/templates/formplayer/case_list.html]: the way to add data to html is to use the `data-` prefix. so this should be `data-index=`

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js [hunk 5]
```diff
@@ -252,8 +258,9 @@ FormplayerFrontend.module("Menus.Views", function (Views, FormplayerFrontend, Ba
             'keypress': 'keyAction',
         },
 
-        caseListAction: function () {
-            FormplayerFrontend.trigger("menu:select", "action 0");
+        caseListAction: function (e) {
+            var index = e.currentTarget.getAttribute('index')
```
Comment (benrudolph) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js]: if you use `data-index` you can then do `$(e.currentTarget).data().index`

**Review by benrudolph — CHANGES_REQUESTED** [review]

**Comment by czue** [comment]
thanks bud

**Comment by wpride** [comment]
@benrudolph thanks - the graph stuff shouldn't have been in here

**Review by benrudolph — APPROVED** [review]
nice 👍

**Pull request merged**